app/src/main/java/cm/aptoide/pt/social/view/TimelineFragment.java

```diff
@@ -614,19 +614,22 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
   }
 
   private void handleSharePreviewAnswer() {
+
+    final ShareEvent[] share = new ShareEvent[1];
     shareDialog.cancels()
-        .doOnNext(shareEvent -> timelineAnalytics.sendShareCompleted(false))
+        .doOnNext(shareEvent -> timelineAnalytics.sendErrorShareCompleted(shareEvent, false, 0))
         .compose(bindUntilEvent(LifecycleEvent.PAUSE))
         .subscribe();
 
     shareDialog.shares()
         .doOnNext(event -> sharePostPublishSubject.onNext(event))
-        .doOnNext(shareEvent -> timelineAnalytics.sendShareCompleted(true))
+        .doOnNext(shareEvent -> timelineAnalytics.sendShareCompleted(shareEvent, true))
         .compose(bindUntilEvent(LifecycleEvent.PAUSE))
         .subscribe(shareEvent -> {
+          share[0] =shareEvent;
         }, throwable -> {
           crashReport.log(throwable);
-          timelineAnalytics.sendShareCompleted(false);
+          timelineAnalytics.sendErrorShareCompleted(share[0], false, 1);
```

Comment:
Is this what we want to log? There should not be errors on the UI when user clicks cancel or share, which is what it looks this is trying to catch.

What we might want to do is follow the sharePostPublishSubject emission and check where is it being subscribed in the TimelinePresenter, and log the error there. This would also clean the 1 element array "hack" that it is used to log errors.